schedule/schedule-appengine/src/main/java/com/jasify/schedule/appengine/model/users/DefaultUserService.java

```diff
@@ -156,102 +160,113 @@ private void notify(User user) {
     }
 
     @Override
-    public User save(User user) throws EntityNotFoundException, FieldValueException {
+    public User save(final User user) throws ModelException {
         //TODO: permissions?
+        TransactionOperator.execute(new ModelOperation<Void>() {
+            @Override
+            public Void execute(com.google.appengine.api.datastore.Transaction tx) throws EntityNotFoundException, FieldValueException {
+                User db = Datastore.getOrNull(tx, userMeta, user.getId());
+                if (db == null) {
+                    throw new EntityNotFoundException();
```

Comment:
Just to "make my point", why bother throwing `EntityNotFoundException` if the signature is `ModelException`